### PR TITLE
Enum from value deriving

### PR DESCRIPTION
**Pull request by paales**

### CODE REVIEW

File: docs/Derivings.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ There are 14 deriving types for now:
 - AggregateChanged
 - Command
 - DomainEvent
-- Enum
+- Enum / Enum(useValue)
```
Comment (prolic) [inline docs/Derivings.md]: Maybe also add Enum(useName) and note in the docs it's the default

File: src/parse.php [hunk 2]
```diff
@@ -439,9 +439,32 @@ function parse(string $filename, array $derivingMap): DefinitionCollection
 
                         $derivingName = $token[1];
 
-                        $derivings[] = $derivingMap[$token[1]];
+                        $derivingType = $derivingMap[$token[1]];
+
                         $token = $skipWhitespace($nextToken());
 
+                        $enumArgs = [];
```
Comment (prolic) [inline src/parse.php]: Can we use this argument parsing also for other derivings?

File: tests/Builder/BuildArgumentConstructorTest.php [hunk 3]
```diff
@@ -169,4 +169,64 @@ public function it_cannot_build_without_any_deriving(): void
 
         buildArgumentConstructor($argument, $definition, $collection);
     }
+
+    /**
+     * @test
+     */
+    public function is_can_build_enum_as_value()
```
Comment (prolic) [inline tests/Builder/BuildArgumentConstructorTest.php]: Return void

File: tests/Builder/BuildToArrayBodyTest.php [hunk 4]
```diff
@@ -90,4 +90,63 @@ public function it_builds_to_array_body(): void
 
         $this->assertSame($expected, buildToArrayBody($definition, $constructor, new DefinitionCollection($definition, $userId, $email), ''));
     }
+
+    /**
+     * @test
+     */
+    public function it_builds_enum_to_body()
```
Comment (prolic) [inline tests/Builder/BuildToArrayBodyTest.php]: Return void

**Comment by paales** [comment]
Yeah! Tests are green :)

**Comment by paales** [comment]
Found a bug in production, lemme see if I can add a test for that as well.

**Comment by prolic** [comment]
Sorry for getting back on this so late, Travis says:

```
  Line   src/helpers.php                                        
 ------ ------------------------------------------------------- 
  346    Call to an undefined method Fpp\Deriving::useValue().  
```

**Comment by prolic** [comment]
ping @paales

**Pull request merged**